### PR TITLE
Add `contains(_:)` methods to (Closed)Range

### LINKED ISSUE #70280: Range.contains(Range) currently checks all values instead of a bounds check, taking a very long time for large ranges.
### Description

The `ClosedRange.contains(ClosedRange)` check on seems to currently inherit from `Collection` which expands out all the values and checks each one. This means large ranges take a very long time to complete a .contains() check. 

### Reproduction

```swift
public extension ClosedRange<Int> {
    func containsCheck(by other: ClosedRange<Int>) -> [ClosedRange<Int>] {
        //print("cropping...")
        
        
         //HANGS when the ranges are large.
         guard !other.contains(self) else {
             print("other contains source entirely. Nothing left over.")
             return []
         }
        
        //Bounds check to the rescue.
        
        guard !other.fullyContains(self) else {
            print("other contains source entirely. Nothing left over.")
            return []
        }
        
        //Already fast.
        if self.contains(other.lowerBound) {
            //print("found lower")
            return [self.lowerBound...other.lowerBound-1]
        } else if self.contains(other.upperBound) {
            //print("found upper")
            return [other.upperBound+1...self.upperBound]
        } else {
            fatalError("Range divide:Forgot something...")
        }
        
    }
    func fullyContains(_ other: ClosedRange<Int>) -> Bool {
        self.lowerBound < other.lowerBound && self.upperBound > other.upperBound
    }
}

let resultC = (2442...2679).containsCheck(by:2321...2446)
print(resultC)

print(Int.max)

let resultD = (2442763622...2679835230).containsCheck(by:2321931404...2446354471)
//let result = (0...9).crop(by:3...8)
print(resultD)

```

### Expected behavior

Range.contains(element) appears to already be optimized as a bounds check. I expected  Range.contains(Range) would have been the same. 

### Environment

swift-driver version: 1.87.1 Apple Swift version 5.9 (swiftlang-5.9.0.128.108 clang-1500.0.40.1)
Target: arm64-apple-macosx14.0

### Additional information

[Forum discussion, discovered as part of Advent of Code 2023](https://forums.swift.org/t/advent-of-code-2023/68749/31)

### PR DESCRIPTION
**Pull request by natecook1000**
The _StringProcessing module provides a generic, collection-based `contains` method that performs poorly for ranges and closed ranges. This addresses the primary issue by providing concrete overloads for Range and ClosedRange which match the expected performance for these operations.

This change also fixes an issue with the existing range overlap tests. The generated `(Closed)Range.overlap` tests are ignoring the "other" range type when generating ranges for testing, so all overlap tests are only being run against ranges of the same type. This fixes things so that heterogeneous testing is included.

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/ClosedRange.swift [hunk 1]
```diff
@@ -325,6 +325,66 @@ where Bound: Strideable, Bound.Stride: SignedInteger
...
+  ///     range.contains(20..<20)   // true
+  ///
+  /// - Parameter other: A range to check for containment within this closed
+  ///   range.
+  /// - Returns: `true` if `other` is empty or wholly contained within this
+  ///   closed range; otherwise, `false`.
+  ///
+  /// - Complexity: O(1)
+  @backDeployed(before: SwiftStdlib 6.1)
+  @inlinable
```
Comment (Azoy) [inline stdlib/public/core/ClosedRange.swift:357]: You can remove both of these attributes and just use `@_alwaysEmitIntoClient`.

**Comment by benrimmington** [comment]
Resolves #70280

**Review by stephentyrone — APPROVED** [review]

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

**Comment by natecook1000** [comment]
@swift-ci Please smoke test macOS platform

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

**Comment by natecook1000** [comment]
@swift-ci Please smoke test Linux platform

File: stdlib/public/core/ClosedRange.swift (hunk 1 above)
Comment (natecook1000) [reply stdlib/public/core/ClosedRange.swift]: Done 👍🏻

**Comment by stephentyrone** [comment]
Can we add a simple benchmark so that a change that hides this overload will get flagged?

**Comment by natecook1000** [comment]
@swift-ci Please benchmark

File: benchmark/single-source/RangeContains.swift [hunk 2]
```diff
@@ -0,0 +1,96 @@
+//===--- RangeOverlaps.swift ----------------------------------------------===//
```
Comment (benrimmington) [inline benchmark/single-source/RangeContains.swift:1]:
```suggestion
//===--- RangeContains.swift ----------------------------------------------===//
```

File: benchmark/single-source/RangeContains.swift [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
```
Comment (benrimmington) [inline benchmark/single-source/RangeContains.swift:5]:
```suggestion
// Copyright (c) 2024 Apple Inc. and the Swift project authors
```

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

**Comment by stephentyrone** [comment]
@swift-ci test Linux

**Pull request merged**